### PR TITLE
feat: [CDS-44423]: Validations on Tab Change - Freeze windows

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx [hunk 1]
```diff
@@ -301,6 +309,20 @@ const ConfigsSection = ({
...
+  React.useEffect(() => {
+    // When we change tab, we should open the invalid rules
+    if (validationErrors?.entity) {
+      const errors = validationErrors.entity || []
+      setEditViews(_editViews => {
+        const newEditViews = [..._editViews]
+        errors.forEach((val, index) => {
+          newEditViews[index] = isEmpty(val) ? newEditViews[index] : true
+        })
+        return newEditViews
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx:321]: could this be done with a map? we anyway aren't doing deepClone here.

Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx:321]: nit: can use map for this

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx [hunk 2]
```diff
@@ -455,3 +485,5 @@ export const FreezeStudioConfigSection: React.FC<FreezeStudioConfigSectionProps>
     </Container>
   )
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx:486]: This File has 5 components, can we extract them and make it one component per file as much as possible?

Component + ForwardRef export, Provider + Immediate consumer should be fine in a single file

**Review by udayvunnam — APPROVED** [review]

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx:321]: Yes agreed, we can use map also for this one.

Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx:321]: map will return a new array of size errors.length
here newEditViews can be of length 5, while errors can be of length 3

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx (hunk 2 above)
Comment (chetanvekariya-harness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/FreezeStudioConfigSection.tsx:486]: +1

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**